packages/engine/Source/Shaders/Builtin/Functions/computeAtmosphereColor.glsl

```diff
@@ -0,0 +1,44 @@
+/**
+ * Compute the atmosphere color, applying Rayleigh and Mie scattering. This
+ * builtin uses automatic uniforms so the atmophere settings are synced with the
+ * state of the Scene, even in other contexts like Model.
+ *
+ * @name czm_computeAtmosphereColor
+ * @glslFunction
+ *
+ * @param {vec3} positionWC Position of the fragment in world coords (low precision)
+ * @param {vec3} lightDirection Light direction from the sun or other light source.
+ * @param {vec3} rayleighColor The Rayleigh scattering color computed by a scattering function
+ * @param {vec3} mieColor The Mie scattering color computed by a scattering function
+ * @param {float} opacity The opacity computed by a scattering function.
+ */
+vec4 czm_computeAtmosphereColor(
```

Comment:
Add a note to #11681 that this can be used in other shaders once the API change happens.

Reply:
Can we eliminate the code duplication even before the API change?
I know at one point you mentioned the Globe using defines rather than a uniform. That seems like a change we could make internally without affecting the API.

Reply:
I certainly would like that but I need to examine how big of a change this would be. The problem is one version of these functions use uniforms set by a specific primitive (`u_atmosphereWhatever`) and others use the new automatic uniforms (`czm_atmosphereWhatever`). You'd need to tie the two together. I have an idea about how to do that, but I have to see if it's too big for this PR.

Reply:
https://github.com/CesiumGS/cesium/issues/11681#issuecomment-1899006593 -- here are more ideas of how to make this work during the deprecation period (in short, have the globe/sky hold a reference to the atmosphere whenever attached to a scene, and the old parameters will just update the new common one). 

While certainly this could be done before the API deprecation, I think it's beyond the scope of this PR since it would add a decent chunk of code here. @ggetz would you agree?

Reply:
Let's try to keep the scope of this PR small if possible.